.github/workflows/nix.yml

```diff
@@ -7,8 +7,12 @@ env:
 
 jobs:
   build:
-
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
```

Comment:
Might want to add `ubuntu-latest-arm-2core` if you want to test on there too (not sure if you need to opt into the beta arm runners somewhere first)

Reply:
The link to the private beta seems to be [invalid/expired](https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/) :(